basic_credential/src/lib.rs

```diff
@@ -75,14 +130,14 @@ fn id(public_key: &[u8], signature_scheme: SignatureScheme) -> Vec<u8> {
     id
 }
 
-impl MlsEntity for SignatureKeyPair {
-    const ID: MlsEntityId = MlsEntityId::SignatureKeyPair;
+impl MlsEntity for OpenMlsBasicCredential {
+    const ID: MlsEntityId = MlsEntityId::OpenMlsBasicCredential;
```

Comment:
Small thing to keep in mind; 

The current state of `main` decorrelates Credentials (whether they're from a Basic credential or a x509 credential) from SignatureKeyPairs (whether again, they're from a basic credential or part of a x509 chain).

This seems to be a step back in the sense that Basic Credentials are again bound to their signature keypair (reminder, there can be N signature keypairs for the same credential, whether it's for ciphersuite reasons, or for the simple fact that the credential itself is a x509 credential and can contain many signature keypairs, up to the user) which seems incorrect in the context of ciphersuite agility.